### ISSUE TITLE
Comparison versus other solutions

### ISSUE THREAD

**Issue by vvo** [baptooo/safely-set-inner-html#13]
Hi! Nice project :)

What are the pro and cons of your solution versus the proposed [StackOverflow solution](https://stackoverflow.com/questions/29044518/safe-alternative-to-dangerouslysetinnerhtml) of using [DOMPurify](https://github.com/cure53/DOMPurify) on the data and then use dangerouslySetInnerHTML?

Shouldn't we purify the data before it even gets into the datastore?

There's also https://github.com/milesj/interweave, how is your module different? Does it brings different features?

Would love to have more details on the context of your project, what was the original issue. Thanks!

**Comment by baptooo** [comment]
Hello @vvo thank you for your report !

>What are the pro and cons of your solution versus the proposed StackOverflow solution of using DOMPurify on the data and then use dangerouslySetInnerHTML?
Shouldn't we purify the data before it even gets into the datastore?

Yes of course the data from your server should be sanitized before being consumed by your client but I wanted to provide a **double-check** solution in order to secure also your client if the job was not
well done or done at all on the server-side.
For example if you work with a customer's data and have no access to its content 😞 

>There's also https://github.com/milesj/interweave, how is your module different? Does it brings different features?

When I started workgin on `safely-set-inner-html` I did a benchmark but I didn't found this solution that looks pretty similar... Thank you for the input, I will take a closer look to it.
Maybe something that I can tell is that it forces you to have a wrapper around your HTML, I am planning to support new [React 16 Fragment](https://reactjs.org/blog/2017/11/28/react-v16.2.0-fragment-support.html).

>Would love to have more details on the context of your project, what was the original issue. Thanks!

So the main idea was to provide this **double-check** thing in case the data you consumes
was not sanitized : 
- customer locking his data
- data from a tier service

Also, `safely-set-inner-html` [gives you feedbacks](https://github.com/baptooo/safely-set-inner-html#blacklist-warnings) about your configuration.
The list will be improved but the idea is to assist the user on his configuration to help him prevent some common XSS cases.

Thanks for your report, do not hesitate if you want to know more 😄
